lsp/src/saros/lsp/SarosLauncher.java

```diff
@@ -1,61 +1,216 @@
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
 import saros.lsp.extensions.client.ISarosLanguageClient;
+import saros.lsp.extensions.server.ISarosLanguageServer;
+import saros.lsp.filesystem.IWorkspacePath;
+import saros.lsp.filesystem.WorkspacePath;
 import saros.lsp.log.LanguageClientAppender;
 
-/** Entry point for the Saros LSP server. */
-public class SarosLauncher {
+/** Entry point for the Saros language server. */
+public class SarosLauncher implements Callable<Integer> {
 
-  private static final Logger log = Logger.getLogger(SarosLauncher.class);
+  private static final Logger LOG = Logger.getLogger(SarosLauncher.class);
+  private static final String LOG4J2_CONFIG_FILENAME = "saros_log4j2.xml";
+
+  @Option(
+      names = {"-p", "--port"},
+      description = "The port to listen on")
+  int port;
+
+  @Option(
+      names = {"-l", "--log"},
+      description = "The log level")
+  String logLevel;
+
+  @Override
+  public Integer call() throws Exception {
+    setupLoggers();
+    LOG.info("listening on port " + port);
+
+    final AsynchronousServerSocketChannel serverSocket =
+        AsynchronousServerSocketChannel.open().bind(new InetSocketAddress("localhost", port));
+
+    final SarosLifecycle lifecycle = new SarosLifecycle();
+
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread() {
+              @Override()
+              public void run() {
+                lifecycle.stop();
+              }
+            });
+
+    lifecycle.start();
+
+    startLanguageServer(lifecycle, serverSocket);
+
+    return 0;
+  }
 
   /**
-   * Starts the server.
+   * Starts the Saros language server.
    *
    * @param args command-line arguments
    * @throws Exception on critical failures
    */
-  public static void main(String[] args) throws Exception {
-
-    if (args.length > 1) {
-      throw new IllegalArgumentException("wrong number of arguments");
-    } else if (args.length != 1) {
-      throw new IllegalArgumentException("port parameter not supplied");
-    } else if (!args[0].matches("\\d+")) {
-      throw new IllegalArgumentException("port is not a number");
-    }
+  public static void main(final String... args) {
+    new CommandLine(new SarosLauncher()).execute(args);
+  }
+
+  /**
+   * Starts the Saros language server.
+   *
+   * @param lifecycle The lifecycle of Saros
+   * @param serverSocket The used socket for a client connection
+   * @throws IOException
+   * @throws InterruptedException
+   * @throws ExecutionException
+   */
+  private void startLanguageServer(
+      final SarosLifecycle lifecycle, final AsynchronousServerSocketChannel serverSocket)
+      throws IOException, InterruptedException, ExecutionException {
+
+    LOG.info("starting saros language server...");
+
+    final ISarosLanguageServer langSvr = lifecycle.getLanguageServer();
+    final AsynchronousSocketChannel socket = createSocket(serverSocket);
+
+    LOG.info("connected to client");
```

Comment:
So the server connects to a client ? Beside that the basic loop of a server is:

```while(true) Socket clientSocket = serverSocket.accept(); /* do something with the clientSocket*/  ```

I am missing this pattern.